install.nsi

```diff
@@ -95,6 +95,35 @@
 ; define language
     !insertmacro MUI_LANGUAGE "English"
 
+;--------------------------------
+; Macro to add firewall inbound rules
+!macro AddFirewallRule RuleName Port
+    ; Create a temporary PowerShell script to add inbound rule
+    FileOpen $0 "$TEMP\${RuleName}_rule.ps1" w
+    FileWrite $0 "$$RuleName = '${RuleName}'$\r$\n"
+    FileWrite $0 "$$Port = ${Port}$\r$\n"
+    FileWrite $0 "$$Protocol = 'TCP'$\r$\n"
+    FileWrite $0 "$$Action = 'Allow'$\r$\n"
+    FileWrite $0 "$$ruleExists = Get-NetFirewallRule -DisplayName $$RuleName -ErrorAction SilentlyContinue$\r$\n"
+    FileWrite $0 "if (-not $$ruleExists) {$\r$\n"
+    FileWrite $0 "    New-NetFirewallRule -DisplayName $$RuleName -Direction Inbound -Protocol $$Protocol -LocalPort $$Port -Action $$Action -Enabled True -Profile Domain,Public$\r$\n"
```

Comment:
This isn't right.   This would only be the case if the computer was configured to think it is on a public network, which nothing we are configuring should be.  If private isn't working, the network mode of the computer needs to change to think it is private, not set it to allow this to public.    Sourcery, make an issue ticket to track adding more help visibility into the current network state, such as if the computer it is installing on thinks it is an office computer.   I would think it should be just private/domain, but def not public.  So if the computer is misconfigured this might not work, but I am OK with that, as the computer should be fixed, not opening it to public networks like if this a laptop that goes to starbucks etc.